File: src/spec/resources/testprojects/nodetest/tests/Api/pet-api.test.ts
import { PetApi } from '../../src/api/pet-api';
import { BearerAuthenticator } from '../../src/auth/bearer-authenticator';
import { Configuration } from '../../src/configuration';
import { Pet } from '../../src/models';

const baseUrl = process.env.API_BASE_URL || 'http://localhost:4010';
const config = new Configuration({
  baseUrl,
  defaultHeaders: { 'Authorization': 'Bearer test-token' },
});
const api = new PetApi(config);
const auth = new BearerAuthenticator(baseUrl, 'test-token');

describe('PetApi', () => {
  test('addPet', async () => {
    const pet: Pet = {
      id: 12345,
      name: 'TestDog',
      photoUrls: ['http://example.com/photo.jpg'],
      status: 'available',
    };

    const result = await api.addPet(auth, pet);

    expect(result).toBeDefined();
    expect(result.name).toBeDefined();
  });

  test('findPetsByStatus', async () => {
    const result = await api.findPetsByStatus('available');

    expect(Array.isArray(result)).toBe(true);
    expect(result.length).toBeGreaterThan(0);
  });

  test('getPetById', async () => {
    const result = await api.getPetById(1);

    expect(result).toBeDefined();
    expect(result.id).toBeDefined();
    expect(result.name).toBeDefined();
  });

  test('updatePet', async () => {
    const pet: Pet = {
      id: 1,
      name: 'UpdatedDog',
      photoUrls: ['http://example.com/updated.jpg'],
      status: 'pending',
    };

    const result = await api.updatePet(1, pet);

    expect(result).toBeDefined();
  });

  test('deletePet', async () => {
    await api.deletePet(auth, 1);
  });
});

File: src/spec/resources/testprojects/csharptest/Tests/Api/PetApiTest.cs
using PetstoreClient;
using PetstoreClient.Api;
using PetstoreClient.Models;
using Xunit;

namespace Tests.Api;

public class PetApiTest
{
    private readonly PetApi _api;

    public PetApiTest()
    {
        var baseUrl = Environment.GetEnvironmentVariable("API_BASE_URL") ?? "http://localhost:4010";
        var config = new Configuration { BaseUrl = baseUrl };
        config.DefaultHeaders["Authorization"] = "Bearer test-token";
        _api = new PetApi(new DefaultApiClient(), config);
    }

    [Fact]
    public async Task TestAddPet()
    {
        var pet = new Pet("TestDog", new List<string> { "http://example.com/photo.jpg" })
        {
            Id = 12345L,
            Status = Pet.StatusEnum.Available,
        };

        var result = await _api.AddPetAsync(pet);

        Assert.NotNull(result);
        Assert.NotNull(result.Name);
    }

    [Fact]
    public async Task TestFindPetsByStatus()
    {
        var result = await _api.FindPetsByStatusAsync("available");

        Assert.NotNull(result);
        Assert.NotEmpty(result);
        Assert.IsType<Pet>(result[0]);
    }

    [Fact]
    public async Task TestGetPetById()
    {
        var result = await _api.GetPetByIdAsync(1L);

        Assert.NotNull(result);
        Assert.NotNull(result.Id);
        Assert.NotNull(result.Name);
    }

    [Fact]
    public async Task TestUpdatePet()
    {
        var pet = new Pet("UpdatedDog", new List<string> { "http://example.com/updated.jpg" })
        {
            Id = 1L,
            Status = Pet.StatusEnum.Pending,
        };

        var result = await _api.UpdatePetAsync(1L, pet);

        Assert.NotNull(result);
    }

    [Fact]
    public async Task TestDeletePet()
    {
        await _api.DeletePetAsync(1L);
        Assert.True(true);
    }
}
